mdm/mdm/unmarshal_json.go

```diff
@@ -4,19 +4,22 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"github.com/google/uuid"
 	"github.com/pkg/errors"
 )
 
 func (c *CommandRequest) UnmarshalJSON(data []byte) error {
 	var request = struct {
 		UDID        string `json:"udid"`
 		RequestType string `json:"request_type"`
-	}{}
+		CommandUUID string `json:"command_uuid"`
+	}{CommandUUID: uuid.New().String()}
```

Comment:
I think the right place for `uuid.New()` is in the `NewCommandPayload()` func. We can just test to see if its empty and fallback to a generated UUID if its empty. I saw your note on UUID working if blank, which of course is hilarious. We _could_ do that check with a string pointer for `CommandUUID`, above, but I'm not sure it's worth the added complexity (expounded a bit on this in #672).

Reply:
I think it's worth it to handle the complexity to check for empty strings. I cannot imagine any utility out of the it being empty, so setting it to the default MicroMDM generated UUID also prevents MicroMDM from having to deal with an  empty string internally that is supposed to be some tangible ID. 

I'll try for that in the simplest way possible.

Reply:
I think we're on the same page here but just to be clear:

> I think it's worth it to handle the complexity to check for empty strings. 

The complexity I refer to here is handling a pointer to a string in order to support usage of an empty UUID which I don't think is worth supporting. This is separate from simply checking for an empty supplied UUID (and defaulting to a generated UUID), which I'm in support of.

The thing here is to separate the concerns of JSON marshaling from default/supplied UUID handling (which probably belongs back in `NewCommandPayload()`).